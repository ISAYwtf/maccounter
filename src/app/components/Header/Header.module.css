.header {
    display: flex;
    align-items: center;
    justify-content: space-between;
}

.headerLeft {
    display: flex;
    align-items: center;
}

.appLogo {
    margin-right: 130px;
}
