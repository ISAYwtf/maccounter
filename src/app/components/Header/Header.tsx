import React from 'react'
import Logo from '@components/Logo/Logo'
import Greeting from '@components/Greeting/Greeting'
import HomePageSwitcher from '@components/HomePageSwitcher/HomePageSwitcher'
import clsx from 'clsx'
import styles from './Header.module.css'

const Header: React.FC = () => (
    <header className={clsx(
        styles.header,
        'app-wrap',
    )}
    >
        <div className={styles.headerLeft}>
            <div className={styles.appLogo}>
                <Logo />
            </div>
            <div>
                <Greeting userName="User" />
            </div>
        </div>
        <HomePageSwitcher />
    </header>
)

export default Header
