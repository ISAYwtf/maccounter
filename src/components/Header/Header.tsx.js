import React from "react"
import styles from "./Header.module.css"
import Logo from "../Logo/Logo.tsx";
import Greeting from "../Greeting/Greeting.tsx";
import HomePageSwitcher from "../HomePageSwitcher/HomePageSwitcher";

const Header = () =>
    <header className={`${styles.header} app-wrap`}>
        <div className={styles.headerLeft}>
            <div className={styles.appLogo}>
                <Logo />
            </div>
            <div>
                <Greeting />
            </div>
        </div>
        <HomePageSwitcher />
    </header>

export default Header
