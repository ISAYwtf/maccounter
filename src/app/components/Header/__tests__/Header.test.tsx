import React from 'react'
import { render } from '@testing-library/react'
import Header from '../Header'

describe('Header', () => {
    const renderComponent = <Header />

    it('render Header', () => {
        const { asFragment } = render(renderComponent)
        expect(asFragment()).toMatchSnapshot()
    })
})
