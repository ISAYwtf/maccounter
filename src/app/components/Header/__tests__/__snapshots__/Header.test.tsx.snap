// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render Header 1`] = `
<DocumentFragment>
  <header
    class="header app-wrap"
  >
    <div
      class="headerLeft"
    >
      <div
        class="appLogo"
      >
        <div
          class="txt--logo"
        >
          <span
            class="txt--orange"
          >
            m.
          </span>
          <span>
            accounter
          </span>
        </div>
      </div>
      <div>
        <div
          class="txt--title-small"
        >
          <p>
            Good morning,Â 
            <span>
              User
            </span>
            !
          </p>
        </div>
      </div>
    </div>
    <div
      class="switcher"
    >
      <p>
        Light mode
      </p>
      <label
        class="switch"
      >
        <input
          type="checkbox"
        />
        <span
          class="slider"
        />
      </label>
    </div>
  </header>
</DocumentFragment>
`;
